feat(k8s): doc autoheal feature (#3027)
scaleway/docs-content: containers/kubernetes/reference-content/using-kubernetes-autoheal-feature.mdx

- **Reduced downtime**: By automatically detecting and recovering from failures, autoheal reduces downtime and minimizes the impact on application performance.
- **Operational efficiency**: It reduces the need for manual intervention in addressing failures, thereby improving operational efficiency.

## Autoheal process

Autoheal is triggered by the shoot-manager every five (5) minutes. If the shoot-manager sees a node in `notReady` status for more than 15 minutes, it will reboot it (once only), and after 30 minutes it will be replaced.

## When to enable or disable autoheal

### When to enable autoheal

Autoheal is advised to be enabled in production environments where maintaining high availability and minimizing downtime is critical.

### When to disable autoheal

There may be scenarios where autoheal should be disabled:

- **Testing environments**: In testing or development environments where failures can be tolerated for troubleshooting purposes.
- **Custom recovery mechanisms**: If custom recovery mechanisms are in place that handle failures more effectively than the autoheal feature.

<Message type="important">

>     You are advised to carefully consider the implications of enabling or disabling autoheal based on your specific use case requirements and operational considerations.

Reviewer: ```suggestion
    We recommend that you carefully consider what enabling or disabling autoheal involves, based on your specific use case requirements and operational considerations.
```
